Feat/docs (#52)
4sure-tech/BC-Wallet-Demo: apps/bc-wallet-demo-web/README.md

# BC Wallet Showcase Demo

## Overview

This application provides a showcase for the BC Wallet to illustrate the use cases for verifiable credentials. This application will take users through multiple steps to demonstrate how verifiable credentials are issued and verified using the BC Wallet.

## Running
### Copy env files

> Copy the files server/.env.example  and client/.env.example to server/.env  and client/.env  

Reviewer: ```suggestion
To set up the development environment, copy the environment configuration files in the following locations:

Navigate to the `apps` directory and copy `.env.example` to `.env` in each of these folders:
- bc-wallet-api-server
- bc-wallet-demo-server
- bc-wallet-demo-web
- bc-wallet-showcase-creator
```

can we create an script for it?

Edit the .env files to match your project needs

### Option 1 - Native

> Please make sure you have a recent version of node, npm, and yarn installed

Reviewer: ```suggestion
Please make sure you have a recent version of node, npm, and pnpm installed
```

These steps are executed from the root folder of the project:

> > yarn install

Reviewer: ```suggestion
> pnpm install
```

> > yarn dev

Reviewer: ```suggestion
> pnpm dev
```

The application will now be running at http://localhost:3000

### Option 2 - Docker
These steps assume that you have docker installed

These steps are executed from the root folder of the project:

Build the client:
> docker build -t bc-wallet-demo-client . -f DockerfileClient

Build the server:
> docker build -t bc-wallet-demo-server . -f DockerfileServer

Start the server:
> docker run --name bc-wallet-demo-server -p5000:5000 --rm --env-file server/.env bc-wallet-demo-server

Start the client:
> docker run --name bc-wallet-demo-client -p3000:3000 -v \`pwd\`/Caddyfile:/etc/caddy/Caddyfile --rm --env-file client/.env bc-wallet-demo-client

The application will now be running at http://localhost:3000

## Contributing

**Pull requests are always welcome!**

Please see the [Contributions Guide](CONTRIBUTING.md) for the repo.

> Before contributing please run `yarn lint --fix` and fix any linter warnings in your code contribution.

Reviewer: ```suggestion
Before contributing please run `pnpm lint --fix` and fix any linter warnings in your code contribution.
```
